content/AlgorytmyWykladniczeZadania.tex

```diff
@@ -0,0 +1,41 @@
+\subsection{Zadanie 3 -- Kolorowanie grafu z list o rozmiarze nie większym niż 2}
+\paragraph{Treść:} W problemie kolorowania grafu $G$ z list dla każdego wierzchołka 
+$v \in V(G)$ mamy daną listę $L_v$ i
+szukamy poprawnego kolorowania grafu $G$, w którym kolor każdego wierzchołka $v$ należy do $L_v$.
+Zaprojektuj algorytm, który rozwiąże problem kolorowania grafu z list w czasie wielomianowym w przypadku, gdy
+wszystkie listy mają rozmiar nie większy niż $2$.
+
+\paragraph{Rozwiązanie:} 
+
+Niech graf $G$ to graf wejściowy w naszym algorytmie, taki, że
+$V(G) = \{v_1, v_2, \ldots, v_n\}$. 
+Niech $\chi = \bigcup_{i = 1}^n L_{v_i}$.
+Utwórzmy zbiór wierzchołków
+\[V' = \{v_{ij} : j \in L_{v_i} \land i \in [n] \land j \in \chi\},\]
+
+oraz zbiór krawędzi 
+\[E' = \{v_{ij}v_{kl} : i = k \land i,k \in [n] \land j, l \in \chi\} \cup 
+\]
+\[\cup
+\{v_{ij}v_{kl} : j = l \land v_iv_k \in E(G) \land i,k \in [n] \land j, l \in \chi\}.\]

```

Comment:
tutaj bym utrzymał poziom formalności tylko pomyślę jak to napisać czytelniej

Reply:
wlasnie to bylo piekielko jesli chodzi o zformalizowanie tego

Reply:
jakbys mial jakis pomysl na to byloby super